### PR TITLE
Fix opting into EngineApi and expect-actual-classes.

### PR DESCRIPTION
**Pull request by colinrtwhite**
`expect-actual-classes` started warning in `1.9.20`.

Pulling this out of https://github.com/cashapp/zipline/pull/1212.

### CODE REVIEW

File: zipline-bytecode/build.gradle.kts [hunk 1]
```diff
@@ -9,10 +9,10 @@ plugins {
 }
 
 kotlin {
-  sourceSets.all {
```
Comment (colinrtwhite) [inline zipline-bytecode/build.gradle.kts:12]: Not sure if this worked previously, but it currently doesn't opt in for the JVM plugin.

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -33,7 +33,6 @@ auto-service-compiler = { module = "dev.zacsweers.autoservice:auto-service-ksp",
 auto-service-annotations = { module = "com.google.auto.service:auto-service-annotations", version = "1.1.1" }
 binary-compatibility-validator-gradle-plugin = { module = "org.jetbrains.kotlinx.binary-compatibility-validator:org.jetbrains.kotlinx.binary-compatibility-validator.gradle.plugin", version = "0.13.2" }
 cklib-gradle-plugin = { module = "co.touchlab:cklib-gradle-plugin", version = "0.3.0" }
-coil-compose = { module = "io.coil-kt:coil-compose", version = "2.5.0" }
```
Comment (colinrtwhite) [inline gradle/libs.versions.toml:36]: Noticed this is unused (unfortunately!)

File: build.gradle.kts [hunk 3]
```diff
@@ -153,6 +153,18 @@ allprojects {
   plugins.withId("org.jetbrains.kotlin.multiplatform") {
     configure<KotlinMultiplatformExtension> {
       jvmToolchain(11)
+      @Suppress("OPT_IN_USAGE")
```
Comment (JakeWharton) [inline build.gradle.kts:156]: Who watches the watchmen?

File: build.gradle.kts [hunk 4]
```diff
@@ -153,6 +153,18 @@ allprojects {
...
+      compilerOptions {
+        freeCompilerArgs.addAll("-opt-in=app.cash.zipline.EngineApi")
+      }
+      // https://youtrack.jetbrains.com/issue/KT-61573
+      targets.configureEach {
+        compilations.configureEach {
+          compilerOptions.configure {
```
Comment (JakeWharton) [inline build.gradle.kts:163]: Surprised you need this last `configure` block. I thought `compilerOptions` returned an object directly on which you could invoke `freeCompilerArgs`.

File: gradle/libs.versions.toml (hunk 2 above)
Comment (JakeWharton) [reply gradle/libs.versions.toml:36]: Ah from emoji probably, before it moved to Redwood.

File: zipline-bytecode/build.gradle.kts (hunk 1 above)
Comment (JakeWharton) [reply zipline-bytecode/build.gradle.kts:12]: Yeah I've only used this construct with the multiplatform plugin. Probably me copy/pasting too aggressively.

**Review by JakeWharton — APPROVED** [review]

File: build.gradle.kts (hunk 4 above)
Comment (colinrtwhite) [reply build.gradle.kts:163]: Interestingly it's a different return type for `KotlinCompilation` and `KotlinProjectExtension`

**Pull request merged**